General Docs Improvements (#1194)
CloudFire-LLC/cloudfire: docs/docs/administer/logs.mdx

---
title: Logs
sidebar_position: 8
description: Firezone's WireGuard®-based remote access platform stores debug logs and configuration, and audit logs.
---

:::note
This guide is written for Docker based deployments of Firezone
:::

> ## What Firezone Logs

Reviewer: Did the capitalization convention change?

Docker deployments of Firezone consist of 3 running containers:

| Container         | Function      | Example logs                                  |
|-------------------|---------------|-----------------------------------------------|
| ubuntu-firezone-1 | Web portal    | HTTP requests received and responses provided |
| ubuntu-postgres-1 | Database      |                                               |
| ubuntu-caddy-1    | Reverse proxy |                                               |

Each container stores logs as a JSON file on the host machine. These files are found at
`var/lib/docker/containers/{CONTAINER_ID}/{CONTAINER_ID}-json.log` and can be shown through the

> `docker logs {CONTAINER}` command.

Reviewer: This section is applicable only to Linux (and looks to be specific to Ubuntu in particular).

It's probably worth rewriting it using `docker compose logs` instead of `docker logs`. Compose abstracts most of the system-specific parts.

Users need to be aware that `docker compose` commands should either be run by within the Firezone root directory at `$HOME/.firezone` or be run with the `-f $HOME/.firezone/docker-compose.yml` flag so that Docker Compose knows where the compose file is.

> ## Managing and Configuring Logs

Reviewer: Capitalization

By default, Firezone uses the `json-file` logging driver without
[additional configuration](https://docs.docker.com/config/containers/logging/json-file/). This means

> logs from each container is stored separately in a file format designed to be exclusively accessed by

Reviewer: ```suggestion
logs from each container are individually stored in a file format designed to be exclusively accessed by
```

the Docker daemon. Log rotation is not enabled, so logs on the host can build up and consume excess
storage space.

For production deployments of Firezone you may want to configure how logs are collected and stored. Docker
provides [multiple mechanisms](https://docs.docker.com/config/containers/logging/configure/)
to collect information from running containers and services.

Examples of popular plugins, configurations, and use cases are:

* Export container logs to
[Splunk](https://docs.docker.com/config/containers/logging/splunk/)
or
[Google Cloud Logging](https://docs.docker.com/config/containers/logging/gcplogs/)
* Enable log rotation and max file size

> * [Customize log driver output](https://docs.docker.com/config/containers/logging/log_tags/)

Reviewer: I think this is pretty neat, and some users will find it useful. Are we advocating users send audit logs (configuration & network activity) to Splunk / GCP as well? Perhaps we should make a clearer distinction.

Author: Since we're not 100% sure how audit logs (config + network activity) are generated and stored at the moment, maybe we can address this later. I put the audit logs section in a separate draft doc.

If they're in the database and require different instructions to view + export, we can add instructions in a separate section with some [Admonitions](https://docusaurus.io/docs/markdown-features/admonitions)? 

I renamed "What Firezone logs" to "Deployment logs".

Reviewer: `Infrastructure logs` or `host logs` or `Debug Logs` *may* be a better term. But yeah, they're going to be completely separate from the STDOUT-based logs managed by Docker. They'll hit the database for sure -- they need to be backed up, redundant, stored with some guarantee of integrity, and possible encrypted. All things supported well by `Ecto.Schema` modules.

> ## Audit Logs

Reviewer: This section probably needs to be its own page and stay in draft format for the time being. Still working through the architecture and implementation.

Author: moved to a draft doc

:::note
Audit logs are in early Beta on the Enterprise plan.
[Contact us](https://www.firezone.dev/contact/sales)
to learn more.
:::

Firezone maintains two types of logs tied to user identity: configuration logs and network activity logs.

Configuration logs track events related to the configuration of Firezone itself, either by admins or users.

> Access logs track connections to protected resources made by users of your network.

Reviewer: `Access logs` is used here but `network activity logs` used in other places -- better to be consistent in marketing, docs, etc.

> Logged events are recorded as JSON objects and stored in the database. These are accessible via a

Reviewer: Just want to clarify that the Docker logs described above are not stored in the database. So we have 3 kinds of logs it seems:

* Infrastructure / deployment logs
* Configuration change logs
* Network activity logs

Thoughts on how to segregate or otherwise draw boundaries between these? I foresee confusion on the horizon.

Author: Having three separate sections on 1 page could work. I changed "What Firezone logs" to "Deployment logs" in the interim before we add config + network activity logs.

We could use [admonitions](https://docusaurus.io/docs/markdown-features/admonitions) for callouts (i.e. enterprise plan, database vs. json storage)

Reviewer: Respectfully disagree. They're going to differ quite substantially in the following ways:

* What they're used for
* Who they're used by (deployment: ICs, audit: CISO / legal / HR)
* How they're used / viewed / accessed
* The integrity level needed
* Where and how they're stored
* Data retention policies / periods

Author: Ok, agreed. Our current logs are more useful for debugging your deployment. Seems more in line with 'troubleshooting' and written for admins.

Let's split out audit logs and put it in 'references' or 'user guides' when we release that feature. It's more of a feature than a deployment detail.  

What do you think?

Reviewer: 👍
